### PR TITLE
test(NODE-3897): re-enable combined coverage

### PR DESCRIPTION
**Pull request by durran**
### Description

Re-enables the combined coverage tasks.

#### What is changing?

Re-enables the tasks and switches to RHEL 8. 

https://mciuploads.s3.amazonaws.com/mongo-node-driver/6fb87e41e6300129503722d6232fcaaf6a556019/6351390d57e85a401615adba/coverage/index.html

##### Is there new documentation needed for these changes?

None

#### What is the motivation for this change?

NODE-3897

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the correct format: `<type>(NODE-xxxx)<!>: <description>`
- [ ] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:875]: If we don't filter out some of the coverage reports then `nyc merge` will fail because the merged coverage object is too large to call `JSON.stringify()` on. Will result in a `RangeError` on this line: https://github.com/istanbuljs/nyc/blob/master/lib/commands/merge.js#L44

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:881]: Without this will fail on `nyc merge` because it does not have `@istanbuljs/nyc-config-typescript` in the environment.

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:870]: This is required until the rhel8 evergreen images have the aws cli installed.

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:877]: aws cli doesn't support regex filtering so a combination of include + exclude is needed. This basically filters out all the erbium/fermium/gallium specific runs.

**Comment by durran** [comment]
https://mciuploads.s3.amazonaws.com/mongo-node-driver/ca51feca25ab1413df7efe9e79b1de051ceb21eb/6345404632f4175d24352219/coverage/index.html

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:883]: Make the the keys in the coverage output relative paths instead of absolute paths.

File: .evergreen/config.in.yml
Comment (durran) [inline .evergreen/config.in.yml:853]: New ticket so getting mongo-node-driver-next as a new bucket doesn't hold up this PR.

**Review by baileympearson — CHANGES_REQUESTED** [review]

File: .evergreen/config.in.yml
Comment (durran) [reply .evergreen/config.in.yml:882]: It is already - problem is npx doesn't install it automatically when running nyc.

File: .evergreen/config.in.yml
Comment (durran) [reply .evergreen/config.in.yml:870]: There is no timeline for them to look at my PR yet so I kept this here in order to move the PR along.

Comment (baileympearson) [reply .evergreen/config.in.yml:870]: Looks like the work is in-progress - https://jira.mongodb.org/browse/BUILD-16116

I think we should wait on the build ticket before merging this PR

**Review by baileympearson — APPROVED** [review]

**Pull request merged**
